src/inc/configuration-document.ts

```diff
@@ -0,0 +1,116 @@
+import defu from 'defu'
+import type { Scheme, OpenIDConnectConfigurationDocument } from '../index'
+import Storage from '../core/storage'
+
+class ConfigurationDocumentRequestError extends Error {
+  constructor() {
+    super('Error loading OpenIDConnect configuration document')
+    this.name = 'ConfigurationDocumentRequestError'
+  }
+}
+
+// eslint-disable-next-line no-console
+const ConfigurationDocumentWarning = (message: string) =>
+  // eslint-disable-next-line no-console
+  console.warn(`[AUTH] [OPENID CONNECT] Invalid configuration. ${message}`)
+
+export default class ConfigurationDocument {
+  public scheme: Scheme
+  public $storage: Storage
+  public key: string
+
+  constructor(scheme: Scheme, storage: Storage) {
+    this.scheme = scheme
+    this.$storage = storage
+    this.key = '_configuration_document.' + this.scheme.name
+  }
+
+  _set(value: OpenIDConnectConfigurationDocument | boolean) {
+    return this.$storage.setState(this.key, value)
```

Comment:
@JoaoPedroAS51 Is this consistent with what we do elsewhere?

Reply:
We should not use `setState` directly. Use `setUniversal` instead.

Reply:
I've replied in your review below.